doc/man/man3/Token.3
.TH "Token" 3 "Fri May 14 2021" "Version 2.1.0" "Text-Classifier" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Token \- This class describes a token\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <token\&.hpp>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBToken\fP (void)"
.br
.RI "Constructs a new instance\&. "
.ti -1c
.RI "\fBToken\fP (std::string name)"
.br
.RI "Constructs a new instance\&. "
.ti -1c
.RI "\fB~Token\fP (void)"
.br
.RI "Destroys the object\&. "
.ti -1c
.RI "std::string \fBget_Name\fP (void) const"
.br
.RI "Gets the name\&. "
.ti -1c
.RI "unsigned \fBget_Ammount\fP (void) const"
.br
.RI "Gets the ammount\&. "
.ti -1c
.RI "float \fBget_Probability\fP (void) const"
.br
.RI "Gets the probability\&. "
.ti -1c
.RI "float \fBget_MultiClass\fP (unsigned pos) const"
.br
.RI "Gets the multi class\&. "
.ti -1c
.RI "std::vector< float > \fBget_MultiClass\fP (void) const"
.br
.RI "Gets the multi class\&. "
.ti -1c
.RI "std::string \fBget_Type\fP (void) const"
.br
.RI "Gets the type\&. "
.ti -1c
.RI "void \fBset_Name\fP (std::string name)"
.br
.RI "Sets the name\&. "
.ti -1c
.RI "void \fBset_Ammount\fP (unsigned ammount)"
.br
.RI "Sets the ammount\&. "
.ti -1c
.RI "void \fBset_Probability\fP (float newProbability)"
.br
.RI "Sets the probability\&. "
.ti -1c
.RI "void \fBset_MultiClass\fP (std::vector< float > newMultiClass)"
.br
.RI "Sets the multi class\&. "
.ti -1c
.RI "void \fBset_Type\fP (std::string newType)"
.br
.RI "Sets the type\&. "
.ti -1c
.RI "bool \fBoperator<\fP (const \fBToken\fP &otherToken) const"
.br
.RI "Less-than comparison operator\&. "
.ti -1c
.RI "bool \fBoperator<=\fP (const \fBToken\fP &otherToken) const"
.br
.RI "Less-than-or-equal comparison operator\&. "
.ti -1c
.RI "bool \fBoperator<\fP (const std::string &str) const"
.br
.RI "Less-than comparison operator\&. "
.ti -1c
.RI "bool \fBoperator<=\fP (const std::string &str) const"
.br
.RI "Less-than-or-equal comparison operator\&. "
.ti -1c
.RI "bool \fBoperator>\fP (const \fBToken\fP &otherToken) const"
.br
.RI "Greater-than comparison operator\&. "
.ti -1c
.RI "bool \fBoperator>=\fP (const \fBToken\fP &otherToken) const"
.br
.RI "Greater-than-or-equal comparison operator\&. "
.ti -1c
.RI "bool \fBoperator>\fP (const std::string &str) const"
.br
.RI "Greater-than comparison operator\&. "
.ti -1c
.RI "bool \fBoperator>=\fP (const std::string &str) const"
.br
.RI "Greater-than-or-equal comparison operator\&. "
.ti -1c
.RI "bool \fBoperator==\fP (const \fBToken\fP &otherToken) const"
.br
.RI "Equality operator\&. "
.ti -1c
.RI "bool \fBoperator==\fP (const std::string &str) const"
.br
.RI "Equality operator\&. "
.ti -1c
.RI "\fBToken\fP & \fBoperator++\fP (void)"
.br
.RI "Increment operator\&. "
.ti -1c
.RI "\fBToken\fP & \fBoperator=\fP (const \fBToken\fP &otherToken)"
.br
.RI "Assignment operator\&. "
.ti -1c
.RI "void \fBincrementate\fP (void)"
.br
.RI "Incremetates the ammount in 1\&. "
.ti -1c
.RI "void \fBaddClassProb\fP (float prob, std::string newClass)"
.br
.RI "Adds the class probability and newClass type if aplicable\&. "
.ti -1c
.RI "float \fBgenerateLogProb\fP (const unsigned &vocSize, const unsigned &tokenAmmount)"
.br
.RI "Generates the ln probability of the token\&. "
.ti -1c
.RI "void \fBprintToken\fP (void) const"
.br
.RI "Prints a token\&. "
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "std::string \fBname_\fP"
.br
.ti -1c
.RI "unsigned \fBammount_\fP"
.br
.ti -1c
.RI "float \fBprobability_\fP"
.br
.ti -1c
.RI "std::vector< float > \fBmultiClass_\fP"
.br
.ti -1c
.RI "std::string \fBtype_\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
This class describes a token\&. 
.PP
Definition at line 34 of file token\&.hpp\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "Token::Token (std::string name)"

.PP
Constructs a new instance\&. 
.PP
\fBParameters\fP
.RS 4
\fIname\fP The name 
.RE
.PP

.PP
Definition at line 43 of file token\&.cpp\&.
.PP
.nf
43                             {
44     set_Name(name);
45     set_Ammount(1);
46     set_Probability(-9999999999999\&.99);
47     set_Type("");
48 }
.fi
.SH "Member Function Documentation"
.PP 
.SS "void Token::addClassProb (float prob, std::string newClass)"

.PP
Adds the class probability and newClass type if aplicable\&. 
.PP
\fBParameters\fP
.RS 4
\fIprob\fP The prob 
.br
\fInewClass\fP The new class 
.RE
.PP

.PP
Definition at line 311 of file token\&.cpp\&.
.PP
.nf
311                                                         {
312     multiClass_\&.push_back(prob);
313     if (prob > probability_) {
314         probability_ = prob;
315         type_ = newClass;
316     }
317 }
.fi
.SS "float Token::generateLogProb (const unsigned & vocSize, const unsigned & tokenAmmount)"

.PP
Generates the ln probability of the token\&. 
.PP
\fBParameters\fP
.RS 4
\fIvocSize\fP The voc size 
.br
\fItokenAmmount\fP The token ammount
.RE
.PP
\fBReturns\fP
.RS 4
The ln probability 
.RE
.PP

.PP
Definition at line 327 of file token\&.cpp\&.
.PP
.nf
327                                                                                    {
328     probability_ = (ammount_ + 1);
329     int divider = (vocSize + tokenAmmount);
330     probability_ /= divider;
331     return std::log(probability_);
332 }
.fi
.SS "unsigned Token::get_Ammount (void) const"

.PP
Gets the ammount\&. 
.PP
\fBReturns\fP
.RS 4
The ammount\&. 
.RE
.PP

.PP
Definition at line 70 of file token\&.cpp\&.
.PP
.nf
70                                        {
71     return ammount_;
72 }
.fi
.SS "float Token::get_MultiClass (unsigned pos) const"

.PP
Gets the multi class\&. 
.PP
\fBParameters\fP
.RS 4
\fIpos\fP The position
.RE
.PP
\fBReturns\fP
.RS 4
The multi class\&. 
.RE
.PP

.PP
Definition at line 90 of file token\&.cpp\&.
.PP
.nf
90                                                {
91     if (pos < multiClass_\&.size()) 
92         return multiClass_[pos];
93     else {
94         std::cout << std::endl << "Error, pos must be lower than multiClass_ size\&." << std::endl;
95         exit(1);
96     }
97 }
.fi
.SS "std::vector< float > Token::get_MultiClass (void) const"

.PP
Gets the multi class\&. 
.PP
\fBReturns\fP
.RS 4
The multi class\&. 
.RE
.PP

.PP
Definition at line 104 of file token\&.cpp\&.
.PP
.nf
104                                                   {
105     return multiClass_;
106 }
.fi
.SS "std::string Token::get_Name (void) const"

.PP
Gets the name\&. 
.PP
\fBReturns\fP
.RS 4
The name\&. 
.RE
.PP

.PP
Definition at line 61 of file token\&.cpp\&.
.PP
.nf
61                                     {
62     return name_;
63 }
.fi
.SS "float Token::get_Probability (void) const"

.PP
Gets the probability\&. 
.PP
\fBReturns\fP
.RS 4
The probability\&. 
.RE
.PP

.PP
Definition at line 79 of file token\&.cpp\&.
.PP
.nf
79                                         {
80     return probability_;
81 }
.fi
.SS "std::string Token::get_Type (void) const"

.PP
Gets the type\&. 
.PP
\fBReturns\fP
.RS 4
The type\&. 
.RE
.PP

.PP
Definition at line 113 of file token\&.cpp\&.
.PP
.nf
113                                      {
114     return type_;
115 }
.fi
.SS "\fBToken\fP & Token::operator++ (void)"

.PP
Increment operator\&. 
.PP
\fBReturns\fP
.RS 4
The result of the increment 
.RE
.PP

.PP
Definition at line 277 of file token\&.cpp\&.
.PP
.nf
277                               {
278     this -> set_Ammount(get_Ammount() + 1);
279     return *this;
280 }
.fi
.SS "bool Token::operator< (const std::string & str) const"

.PP
Less-than comparison operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIstr\fP The string
.RE
.PP
\fBReturns\fP
.RS 4
The result of the less-than comparison 
.RE
.PP

.PP
Definition at line 191 of file token\&.cpp\&.
.PP
.nf
191                                                  {
192     return (get_Name() < str);
193 }
.fi
.SS "bool Token::operator< (const \fBToken\fP & otherToken) const"

.PP
Less-than comparison operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIotherToken\fP The other token
.RE
.PP
\fBReturns\fP
.RS 4
The result of the less-than comparison 
.RE
.PP

.PP
Definition at line 169 of file token\&.cpp\&.
.PP
.nf
169                                                     {
170     return (get_Name() < otherToken\&.get_Name());
171 }
.fi
.SS "bool Token::operator<= (const std::string & str) const"

.PP
Less-than-or-equal comparison operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIstr\fP The string
.RE
.PP
\fBReturns\fP
.RS 4
The result of the less-than-or-equal comparison 
.RE
.PP

.PP
Definition at line 202 of file token\&.cpp\&.
.PP
.nf
202                                                   {
203     return (get_Name() <= str);
204 }
.fi
.SS "bool Token::operator<= (const \fBToken\fP & otherToken) const"

.PP
Less-than-or-equal comparison operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIotherToken\fP The other token
.RE
.PP
\fBReturns\fP
.RS 4
The result of the less-than-or-equal comparison 
.RE
.PP

.PP
Definition at line 180 of file token\&.cpp\&.
.PP
.nf
180                                                      {
181     return (get_Name() <= otherToken\&.get_Name());
182 }
.fi
.SS "\fBToken\fP & Token::operator= (const \fBToken\fP & otherToken)"

.PP
Assignment operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIotherToken\fP The other token
.RE
.PP
\fBReturns\fP
.RS 4
The result of the assignment 
.RE
.PP

.PP
Definition at line 289 of file token\&.cpp\&.
.PP
.nf
289                                                 {
290     this -> set_Name(otherToken\&.get_Name());
291     this -> set_Ammount(otherToken\&.get_Ammount());
292     this -> set_Probability(otherToken\&.get_Probability());
293     this -> set_Type(otherToken\&.get_Type());
294     this -> set_MultiClass(otherToken\&.get_MultiClass());
295     return *this;
296 }
.fi
.SS "bool Token::operator== (const std::string & str) const"

.PP
Equality operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIstr\fP The string
.RE
.PP
\fBReturns\fP
.RS 4
The result of the equality 
.RE
.PP

.PP
Definition at line 268 of file token\&.cpp\&.
.PP
.nf
268                                                   {
269     return (get_Name() == str);
270 }
.fi
.SS "bool Token::operator== (const \fBToken\fP & otherToken) const"

.PP
Equality operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIotherToken\fP The other token
.RE
.PP
\fBReturns\fP
.RS 4
The result of the equality 
.RE
.PP

.PP
Definition at line 257 of file token\&.cpp\&.
.PP
.nf
257                                                      {
258     return (get_Name() == otherToken\&.get_Name());
259 }
.fi
.SS "bool Token::operator> (const std::string & str) const"

.PP
Greater-than comparison operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIstr\fP The string
.RE
.PP
\fBReturns\fP
.RS 4
The result of the greater-than comparison 
.RE
.PP

.PP
Definition at line 235 of file token\&.cpp\&.
.PP
.nf
235                                                  {
236     return (get_Name() > str);
237 }
.fi
.SS "bool Token::operator> (const \fBToken\fP & otherToken) const"

.PP
Greater-than comparison operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIotherToken\fP The other token
.RE
.PP
\fBReturns\fP
.RS 4
The result of the greater-than comparison 
.RE
.PP

.PP
Definition at line 213 of file token\&.cpp\&.
.PP
.nf
213                                                     {
214     return (get_Name() > otherToken\&.get_Name());
215 }
.fi
.SS "bool Token::operator>= (const std::string & str) const"

.PP
Greater-than-or-equal comparison operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIstr\fP The string
.RE
.PP
\fBReturns\fP
.RS 4
The result of the greater-than-or-equal comparison 
.RE
.PP

.PP
Definition at line 246 of file token\&.cpp\&.
.PP
.nf
246                                                   {
247     return (get_Name() >= str);
248 }
.fi
.SS "bool Token::operator>= (const \fBToken\fP & otherToken) const"

.PP
Greater-than-or-equal comparison operator\&. 
.PP
\fBParameters\fP
.RS 4
\fIotherToken\fP The other token
.RE
.PP
\fBReturns\fP
.RS 4
The result of the greater-than-or-equal comparison 
.RE
.PP

.PP
Definition at line 224 of file token\&.cpp\&.
.PP
.nf
224                                                      {
225     return (get_Name() >= otherToken\&.get_Name());
226 }
.fi
.SS "void Token::set_Ammount (unsigned ammount)"

.PP
Sets the ammount\&. 
.PP
\fBParameters\fP
.RS 4
\fIammount\fP The ammount 
.RE
.PP

.PP
Definition at line 131 of file token\&.cpp\&.
.PP
.nf
131                                          {
132     ammount_ = ammount;
133 }
.fi
.SS "void Token::set_MultiClass (std::vector< float > newMultiClass)"

.PP
Sets the multi class\&. 
.PP
\fBParameters\fP
.RS 4
\fInewMultiClass\fP The new multi class 
.RE
.PP

.PP
Definition at line 149 of file token\&.cpp\&.
.PP
.nf
149                                                           {
150     multiClass_ = newMultiClass;
151 }
.fi
.SS "void Token::set_Name (std::string name)"

.PP
Sets the name\&. 
.PP
\fBParameters\fP
.RS 4
\fIname\fP The name 
.RE
.PP

.PP
Definition at line 122 of file token\&.cpp\&.
.PP
.nf
122                                     {
123     name_ = name;
124 }
.fi
.SS "void Token::set_Probability (float newProbability)"

.PP
Sets the probability\&. 
.PP
\fBParameters\fP
.RS 4
\fInewProbability\fP The new probability 
.RE
.PP

.PP
Definition at line 140 of file token\&.cpp\&.
.PP
.nf
140                                                  {
141     probability_ = newProbability;
142 }
.fi
.SS "void Token::set_Type (std::string newType)"

.PP
Sets the type\&. 
.PP
\fBParameters\fP
.RS 4
\fInewType\fP The new type 
.RE
.PP

.PP
Definition at line 158 of file token\&.cpp\&.
.PP
.nf
158                                        {
159     type_ = newType;
160 }
.fi


.SH "Author"
.PP 
Generated automatically by Doxygen for Text-Classifier from the source code\&.


src/token.cpp
/*=======================================================================================
=========================================================================================
	=                                                                              =
	=            Proyect:       Text Classifier                                    =
	=            File name:     token.cpp                                          =
	=            Author:        Adrián Epifanio Rodríguez Hernández                =
	=            Date:          21/04/2021                                         =
	=            Subject:       Advanced Artificial Inteligence                    =
	=            Language:      C++                                                =
	=            Email:         alu0101158280@ull.edu.es                           =
	=            Place:         Universidad De La Laguna                           =
	=                           Escuela Superior de Ingeniería y Tecnología        =
	=                                                                              =
=========================================================================================
=======================================================================================*/
/*
* @Author: Adrian Epifanio
* @Date:   2021-04-25 21:00:43
* @Last Modified by:   Adrian Epifanio
* @Last Modified time: 2021-05-11 23:39:28
*/
/*------------------  FUNCTIONS  -----------------*/

#include "../include/token.hpp"

/*------------------------------------------------*/

/**
 * @brief      Constructs a new instance.
 */
Token::Token (void) {
	set_Name("");
	set_Ammount(0);
	set_Probability(0);
	set_Type("");
}

/**
 * @brief      Constructs a new instance.
 *
 * @param[in]  name  The name
 */
Token::Token (std::string name) {
	set_Name(name);
	set_Ammount(1);
	set_Probability(-9999999999999.99);
	set_Type("");
}

/**
 * @brief      Destroys the object.
 */
Token::~Token (void) {
}

/**
 * @brief      Gets the name.
 *
 * @return     The name.
 */
std::string Token::get_Name(void) const {
	return name_;
}

/**
 * @brief      Gets the ammount.
 *
 * @return     The ammount.
 */
unsigned Token::get_Ammount (void) const {
	return ammount_;
}

/**
 * @brief      Gets the probability.
 *
 * @return     The probability.
 */
float Token::get_Probability (void) const {
	return probability_;
}

/**
 * @brief      Gets the multi class.
 *
 * @param[in]  pos   The position
 *
 * @return     The multi class.
 */
float Token::get_MultiClass (unsigned pos) const {
	if (pos < multiClass_.size()) 
		return multiClass_[pos];
	else {
		std::cout << std::endl << "Error, pos must be lower than multiClass_ size." << std::endl;
		exit(1);
	}
}

/**
 * @brief      Gets the multi class.
 *
 * @return     The multi class.
 */
std::vector<float> Token::get_MultiClass (void) const {
	return multiClass_;
}

/**
 * @brief      Gets the type.
 *
 * @return     The type.
 */
std::string Token::get_Type (void) const {
	return type_;
}

/**
 * @brief      Sets the name.
 *
 * @param[in]  name  The name
 */
void Token::set_Name (std::string name) {
	name_ = name;
}

/**
 * @brief      Sets the ammount.
 *
 * @param[in]  ammount  The ammount
 */
void Token::set_Ammount (unsigned ammount) {
	ammount_ = ammount;
}

/**
 * @brief      Sets the probability.
 *
 * @param[in]  newProbability  The new probability
 */
void Token::set_Probability (float newProbability) {
	probability_ = newProbability;
}

/**
 * @brief      Sets the multi class.
 *
 * @param[in]  newMultiClass  The new multi class
 */
void Token::set_MultiClass (std::vector<float> newMultiClass) {
	multiClass_ = newMultiClass;
}

/**
 * @brief      Sets the type.
 *
 * @param[in]  newType  The new type
 */
void Token::set_Type (std::string newType) {
	type_ = newType;
}

/**
 * @brief      Less-than comparison operator.
 *
 * @param[in]  otherToken  The other token
 *
 * @return     The result of the less-than comparison
 */
bool Token::operator< (const Token& otherToken) const {
	return (get_Name() < otherToken.get_Name());
}

/**
 * @brief      Less-than-or-equal comparison operator.
 *
 * @param[in]  otherToken  The other token
 *
 * @return     The result of the less-than-or-equal comparison
 */
bool Token::operator<= (const Token& otherToken) const {
	return (get_Name() <= otherToken.get_Name());
}

/**
 * @brief      Less-than comparison operator.
 *
 * @param[in]  str   The string
 *
 * @return     The result of the less-than comparison
 */
bool Token::operator< (const std::string& str) const {
	return (get_Name() < str);
}

/**
 * @brief      Less-than-or-equal comparison operator.
 *
 * @param[in]  str   The string
 *
 * @return     The result of the less-than-or-equal comparison
 */
bool Token::operator<= (const std::string& str) const {
	return (get_Name() <= str);
}

/**
 * @brief      Greater-than comparison operator.
 *
 * @param[in]  otherToken  The other token
 *
 * @return     The result of the greater-than comparison
 */
bool Token::operator> (const Token& otherToken) const {
	return (get_Name() > otherToken.get_Name());
}

/**
 * @brief      Greater-than-or-equal comparison operator.
 *
 * @param[in]  otherToken  The other token
 *
 * @return     The result of the greater-than-or-equal comparison
 */
bool Token::operator>= (const Token& otherToken) const {
	return (get_Name() >= otherToken.get_Name());
}

/**
 * @brief      Greater-than comparison operator.
 *
 * @param[in]  str   The string
 *
 * @return     The result of the greater-than comparison
 */
bool Token::operator> (const std::string& str) const {
	return (get_Name() > str);
}

/**
 * @brief      Greater-than-or-equal comparison operator.
 *
 * @param[in]  str   The string
 *
 * @return     The result of the greater-than-or-equal comparison
 */
bool Token::operator>= (const std::string& str) const {
	return (get_Name() >= str);
}

/**
 * @brief      Equality operator.
 *
 * @param[in]  otherToken  The other token
 *
 * @return     The result of the equality
 */
bool Token::operator== (const Token& otherToken) const {
	return (get_Name() == otherToken.get_Name());
}

/**
 * @brief      Equality operator.
 *
 * @param[in]  str   The string
 *
 * @return     The result of the equality
 */
bool Token::operator== (const std::string& str) const {
	return (get_Name() == str);
}

/**
 * @brief      Increment operator.
 *
 * @return     The result of the increment
 */
Token& Token::operator++ (void) {
	this -> set_Ammount(get_Ammount() + 1);
	return *this;
}

/**
 * @brief      Assignment operator.
 *
 * @param[in]  otherToken  The other token
 *
 * @return     The result of the assignment
 */
Token& Token::operator= (const Token& otherToken) {
	this -> set_Name(otherToken.get_Name());
	this -> set_Ammount(otherToken.get_Ammount());
	this -> set_Probability(otherToken.get_Probability());
	this -> set_Type(otherToken.get_Type());
	this -> set_MultiClass(otherToken.get_MultiClass());
	return *this;
}

/**
 * @brief      Incremetates the ammount in 1.
 */
void Token::incrementate (void) {
	ammount_++;
}

/**
 * @brief      Adds the class probability and newClass type if aplicable.
 *
 * @param[in]  prob      The prob
 * @param[in]  newClass  The new class
 */
void Token::addClassProb (float prob, std::string newClass) {
	multiClass_.push_back(prob);
	if (prob > probability_) {
		probability_ = prob;
		type_ = newClass;
	}
}

/**
 * @brief      Generates the ln probability of the token.
 *
 * @param[in]  vocSize       The voc size
 * @param[in]  tokenAmmount  The token ammount
 *
 * @return     The ln probability
 */
float Token::generateLogProb (const unsigned& vocSize, const unsigned& tokenAmmount) {
	probability_ = (ammount_ + 1);
	int divider = (vocSize + tokenAmmount);
	probability_ /= divider;
	return std::log(probability_);
}

/**
 * @brief      Prints a token.
 */
void Token::printToken (void) const {
	std::cout << std::endl << "Token Info:" << std::endl << "\t - Name: " << get_Name() << std::endl << "\t - Ammount: " << get_Ammount() << std::endl << "\t - Type: " << get_Type() << std::endl << "\t - Prob: " << get_Probability();
}
